File: src/frontend/fileshare-web/src/app/shared/models/file-contracts.ts
export interface TransferProofContract {
  fileHash: string;
  blockNumber: number;
  blockHash: string;
  signature: string;
  issuedAt: string;
}

export interface UploadFileResponse {
  id: string;
  accessToken: string;
  fileName: string;
  expiresAt: string;
  maxDownloads: number | null;
  hasPassword: boolean;
  proof: TransferProofContract;
  metadataUrl: string;
  availabilityUrl: string;
  downloadUrl: string;
}

export interface FileMetadataResponse {
  id: string;
  fileName: string;
  contentType: string;
  size: number;
  createdAt: string;
  expiresAt: string;
  downloadCount: number;
  maxDownloads: number | null;
  status: string;
  hasPassword: boolean;
  proof: TransferProofContract;
}

export interface FileAvailabilityResponse {
  available: boolean;
  status: string;
  reason: string | null;
  expiresAt: string | null;
  downloadCount: number | null;
  maxDownloads: number | null;
  hasPassword: boolean;
}

File: src/backend/FileShare.Contracts/Files/FileContracts.cs
namespace FileShare.Contracts.Files;

public sealed record TransferProofContract(
    string FileHash,
    long BlockNumber,
    string BlockHash,
    string Signature,
    DateTimeOffset IssuedAt);

public sealed record MalwareScanContract(
    string Status,
    int? MaliciousCount,
    int? SuspiciousCount,
    int? TotalEngines,
    DateTimeOffset? ScannedAt,
    string? Permalink,
    bool IsEmulated);

public sealed record UploadFileResponse(
    Guid Id,
    string AccessToken,
    string FileName,
    DateTimeOffset ExpiresAt,
    int? MaxDownloads,
    bool HasPassword,
    TransferProofContract Proof,
    MalwareScanContract Scan,
    string MetadataUrl,
    string AvailabilityUrl,
    string DownloadUrl);

public sealed record FileMetadataResponse(
    Guid Id,
    string FileName,
    string ContentType,
    long Size,
    DateTimeOffset CreatedAt,
    DateTimeOffset ExpiresAt,
    int DownloadCount,
    int? MaxDownloads,
    string Status,
    bool HasPassword,
    TransferProofContract Proof,
    MalwareScanContract Scan);

public sealed record FileAvailabilityResponse(
    bool Available,
    string Status,
    string? Reason,
    DateTimeOffset? ExpiresAt,
    int? DownloadCount,
    int? MaxDownloads,
    bool HasPassword);

public sealed record VerifyProofResponse(
    bool Verified,
    string FileName,
    long Size,
    string FileHash,
    long BlockNumber,
    string BlockHash,
    string Signature,
    DateTimeOffset IssuedAt,
    DateTimeOffset ExpiresAt,
    string Status,
    int DownloadCount,
    int? MaxDownloads);
